src/software/jetson_nano/services/motor.cpp

```diff
@@ -258,13 +292,199 @@ uint8_t MotorService::readWriteByte(uint8_t motor, uint8_t data, uint8_t last_tr
+    writeToControllerOrDieTrying(motor, TMC4671_PID_VELOCITY_LIMIT, 10000);
+}
+
+void MotorService::configureADC(uint8_t motor)
+{
+    // ADC configuration
+    writeToControllerOrDieTrying(motor, TMC4671_ADC_I_SELECT, 0x18000100);
+    writeToControllerOrDieTrying(motor, TMC4671_dsADC_MDEC_B_MDEC_A, 0x014E014E);
+    writeToControllerOrDieTrying(motor, TMC4671_ADC_I0_SCALE_OFFSET, 0x010081DD);
+    writeToControllerOrDieTrying(motor, TMC4671_ADC_I1_SCALE_OFFSET, 0x0100818E);
+}
+
+void MotorService::configureEncoder(uint8_t motor)
+{
+    // ABN encoder settings
+    writeToControllerOrDieTrying(motor, TMC4671_ABN_DECODER_MODE, 0x00000000);
+    writeToControllerOrDieTrying(motor, TMC4671_ABN_DECODER_PPR, 0x00001000);
+}
+
+void MotorService::calibrateEncoder(uint8_t motor)
+{
+    LOG(WARNING) << "Calibrating the encoder, ensure the robot is lifted off the ground";
+    sleep(1);
+
+    writeToControllerOrDieTrying(motor, TMC4671_PID_TORQUE_FLUX_LIMITS, 0x000003E8);
+    writeToControllerOrDieTrying(motor, TMC4671_PID_TORQUE_P_TORQUE_I, 0x01000100);
+    writeToControllerOrDieTrying(motor, TMC4671_PID_FLUX_P_FLUX_I, 0x01000100);
+
+    writeToControllerOrDieTrying(motor, TMC4671_MODE_RAMP_MODE_MOTION, 0x00000008);
+    writeToControllerOrDieTrying(motor, TMC4671_ABN_DECODER_PHI_E_PHI_M_OFFSET,
+                                 0x00000000);
+    writeToControllerOrDieTrying(motor, TMC4671_PHI_E_SELECTION, 0x00000001);
+    writeToControllerOrDieTrying(motor, TMC4671_PHI_E_EXT, 0x00000000);
+    writeToControllerOrDieTrying(motor, TMC4671_UQ_UD_EXT, 0x000007D0);
+    sleep(1);
+
+    writeToControllerOrDieTrying(motor, TMC4671_ABN_DECODER_COUNT, 0x00000000);
+    writeToControllerOrDieTrying(motor, TMC4671_UQ_UD_EXT, 0x00000000);
+    writeToControllerOrDieTrying(motor, TMC4671_PHI_E_SELECTION, TMC4671_PHI_E_ABN);
+
+    encoder_calibrated_[motor] = true;
+}
+
+void MotorService::runOpenLoopCalibrationRoutine(uint8_t motor, size_t num_samples)
+{
+    // Some limits
+    tmc4671_writeInt(motor, TMC4671_PID_TORQUE_FLUX_LIMITS, 0x000003E8);
+    tmc4671_writeInt(motor, TMC4671_PID_TORQUE_P_TORQUE_I, 0x01000100);
+    tmc4671_writeInt(motor, TMC4671_PID_FLUX_P_FLUX_I, 0x01000100);
+
+    // Open loop settings
+    tmc4671_writeInt(motor, TMC4671_OPENLOOP_MODE, 0x00000000);
+    tmc4671_writeInt(motor, TMC4671_OPENLOOP_ACCELERATION, 0x0000003C);
+    tmc4671_writeInt(motor, TMC4671_OPENLOOP_VELOCITY_TARGET, 0xFFFFFFFB);
+
+    // Feedback selection
+    tmc4671_writeInt(motor, TMC4671_PHI_E_SELECTION, 0x00000002);
+    tmc4671_writeInt(motor, TMC4671_UQ_UD_EXT, 0x00000779);
+
+    // Switch to open loop velocity mode
+    tmc4671_writeInt(motor, TMC4671_MODE_RAMP_MODE_MOTION, 0x00000008);
+
+    // Rotate right
+    tmc4671_writeInt(motor, TMC4671_OPENLOOP_VELOCITY_TARGET, 0x0000004A);
+
+    // Setup CSVs
+    LOG(CSV, "encoder_calibration.csv") << "actual_encoder,estimated_phi\n";
```

Comment:
confused about this because there was a comment in another file saying dont log CSV

Reply:
this will go through the CSV sink, so it eventually gets logged to a file, but  it doesn't log through the colored cout sink (so it doesn't spam)